ibc/host/prefix.go

```diff
@@ -0,0 +1,19 @@
+package host
+
+import (
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+)
+
+// ApplyPrefix applies the prefix to the provided path returning a CommitmentPath
+func ApplyPrefix(prefix *coreTypes.CommitmentPrefix, path string) *coreTypes.CommitmentPath {
+	bz := make([]byte, 0, len(prefix.Prefix)+1+len([]byte(path)))
```

Comment:
Have you considered using `strings.Builder` or `bytes.Buffer ` instead of appending bytes? It's a micro optimization but more idiomatic approaches to `Write`ing bytes and strings

Reply:
I think because we assign the capacity to the byte slice the append is actually faster try running this albeit not complex benchmark and see:

```go
package main

import (
	"bytes"
	"fmt"
	"time"
)

func main() {
	start := time.Now()
	bz := make([]byte, 0, 100)
	for i := 0; i < 100000000; i++ {
		bz = append(bz, []byte("a")...)
	}
	elapsed := time.Since(start)
	fmt.Println(elapsed)

	start = time.Now()
	buf := new(bytes.Buffer)
	for i := 0; i < 100000000; i++ {
		buf.Write([]byte("a"))
	}
	elapsed = time.Since(start)
	fmt.Println(elapsed)
}
```
```
$ go run main.go
257.265989ms
439.87575ms
$ go run main.go
162.618257ms
448.861254ms
$ go run main.go
166.865572ms
436.184737ms
 ```